dev/dev-env-setup.sh

```diff
@@ -0,0 +1,233 @@
+      echo "$USERPROFILE\.pyenv\pyenv-win\\bin\\" >> "$GITHUB_PATH"
+    else
+      echo "Unknown operating system environment: $machine exiting."
+      exit 1
+    fi
+  else
+    if [[ "$machine" == win ]]; then
+      PYENV_README=https://github.com/pyenv-win/pyenv-win/blob/master/README.md
+    elif [[ "$machine" == mac || "$machine" == linux ]]; then
+      PYENV_README=https://github.com/pyenv/pyenv/blob/master/README.md
+    else
+      echo "The current OS is unknown. Please visit: https://github.com/pyenv/pyenv#installation for instructions. "
+      exit 1
+    fi
+    echo "pyenv is required to use this environment setup script. Please install by following instructions here: $PYENV_README"
+    exit 1
+  fi
+fi
+
+MLFLOW_HOME=$(pwd)
+
+# Get the minimum supported version from MLflow to ensure any feature development adheres to legacy Python versions
+min_py_version=$(grep "python_requires=" "$MLFLOW_HOME/setup.py" | grep -E -o "([0-9]{1,}\.)+[0-9]{1,}")
+
+echo "The minimum version of Python to ensure backwards compatibility for MLflow development is: $(tput bold; tput setaf 3)$min_py_version$(tput sgr0)"
+
+# Resolve a minor version to the latest micro version
+case $min_py_version in
+  "3.7") PY_INSTALL_VERSION="3.7.13" ;;
+  "3.8") PY_INSTALL_VERSION="3.8.13" ;;
+  "3.9") PY_INSTALL_VERSION="3.9.11" ;;
+  "3.10") PY_INSTALL_VERSION="3.10.3" ;;
+esac
+
+# Install the Python version if it cannot be found
+pyenv install -s $PY_INSTALL_VERSION
+pyenv local $PY_INSTALL_VERSION
+pyenv exec pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) --upgrade pip
+pyenv exec pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) virtualenv
+
+VENV_DIR="$directory/bin/activate"
+
+# Check if the virtualenv already exists at the specified path
+if [[ -d "$directory"  ]]; then
+  if [ -z "$MLFLOW_DEV_ENV_REPLACE_ENV" ]; then
+    read -p "A virtual environment is already located at $directory. Do you wish to replace it? $(tput bold; tput setaf 2)(y/n) $(tput sgr0)" -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || $MLFLOW_DEV_ENV_REPLACE_ENV == 1 ]]; then
+    deactivate
+    rm -rf "$directory"
+    echo "Virtual environment removed from '$directory'. Installing new instance."
+    pyenv exec virtualenv "$directory"
+  fi
+else
+  # Create a virtual environment with the specified Python version
+  pyenv exec virtualenv "$directory"
+fi
+
+# Activate the virtual environment
+# shellcheck disable=SC1090
+source "$VENV_DIR"
+
+echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
+echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
+
+echo "Installing pip dependencies for development environment."
+
+if [[ $full_install == 1 ]]; then
+  # Install required dependencies for Prophet
+  tmp_dir=$(mktemp -d)
+  pip download $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) --no-deps --dest "$tmp_dir" --no-cache-dir prophet
```

Comment:
Can we put `(( quiet == 1 && verbose == 0 )) && printf %s '-q'` in a variable and reuse it?